internal/pkg/service/stream/storage/node/coordinator/filerotation/operator.go

```diff
@@ -150,19 +152,27 @@ func Start(d dependencies, config targetConfig.OperatorConfig) error {
 
 				if restart {
 					o.openedSlicesCount = make(map[model.FileKey]int)
+					o.filesNotEmpty = make(map[model.FileKey]bool)
 				}
 
 				// Update opened slices counts, per file
 				for _, event := range events {
+					fileKey := event.Value.FileKey
 					switch event.Type {
 					case etcdop.CreateEvent:
-						o.openedSlicesCount[event.Value.FileKey]++
+						o.openedSlicesCount[fileKey]++
+						if event.Value.State == model.SliceUploading {
+							o.filesNotEmpty[fileKey] = o.filesNotEmpty[fileKey] || !event.Value.LocalStorage.IsEmpty
```

Comment:
`filesNotEmpty` :thinking: 
- suggestions:
- `fileNotEmpty`
- `slicesNotEmpty`

Reply:
Maybe it could be 2 switch statementst, each with a comment what it is doing.